src/main/java/join_room_use_case/JoinByIDInteractor.java

```diff
@@ -0,0 +1,39 @@
+package join_room_use_case;
+
+import signup_and_login.UserDBGateway;
+import signup_and_login.UserDBModel;
+
+import java.util.List;
+
+public class JoinByIDInteractor implements JoinByIDInputBoundary{
+    private final RoomDBGateway roomDBGateway;
+    private final UserDBGateway userDBGateway;
+    private final JoinByIDOutputBoundary presenter;
+
+    public JoinByIDInteractor(RoomDBGateway roomDBGateway, JoinByIDOutputBoundary presenter,
+                              UserDBGateway userDBGateway){
+        this.roomDBGateway = roomDBGateway;
+        this.userDBGateway = userDBGateway;
+        this.presenter = presenter;
+    }
+
+    @Override
+    public void handleJoinByID(JoinByIDRequestModel requestModel){
+        List<RoomDBModel> rooms = roomDBGateway.loadRooms();
+
+        for (RoomDBModel room : rooms) {
+            if (room.getName().equals(requestModel.getRoomName())){
+                List<UserDBModel> users = userDBGateway.loadUsers();
+                for (UserDBModel user : users){
+                    if (user.getUserID() == requestModel.getCurrUserID()){
```

Comment:
I think it is better to use equals() in if (user.getUserID() == requestModel.getCurrUserID())

Reply:
can't use `equals()` in this case since the type of `user.getUserID()` is primitive not object. And since it is primitive, i think == operator is enough. Let me know if I missed something!